vnext/JSI/Shared/ChakraObjectRef.h

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "jsi/jsi.h"
+
+#ifdef CHAKRACORE
+#include "ChakraCore.h"
+#else
+#ifndef USE_EDGEMODE_JSRT
+#define USE_EDGEMODE_JSRT
+#endif
+#include <jsrt.h>
+#endif
+
+#include <exception>
+#include <string>
+
+namespace Microsoft::JSI {
+
+inline void CrashUponChakraError(JsErrorCode error) {
+  if (error != JsNoError) {
+    std::terminate();
+  }
+}
+
+void ThrowUponChakraError(
+    JsErrorCode error,
+    const char *const chakraApiName = nullptr);
```

Comment:
>const char *const chakraApiName = nullptr [](start = 4, length = 41)

Similar to previous comment, this info doesn't seem useful. Once we are throwing, there should be callstack, so it should be easy to determine failing chakra function from that callstack.

Reply:
You are correct. I'll remove this string from the API.

---
In reply to: [336157571](https://github.com/microsoft/react-native-windows/pull/3445#discussion_r336157571) [](ancestors = 336157571)